protostar/cheatable_starknet/callable_hint_locals/declare_hint_local.py

```diff
@@ -1,54 +1,65 @@
+    compute_compiled_class_hash,
+)
+from starkware.starknet.services.api.contract_class.contract_class import ContractClass
+
+from protostar.cairo.cairo_bindings import compile_starknet_contract_from_path
+from protostar.cheatable_starknet.callable_hint_locals.callable_hint_local import (
+    CallableHintLocal,
+)
+from protostar.cheatable_starknet.controllers.contract_class_utils import parse_casm
 from protostar.cheatable_starknet.controllers.contracts import (
-    ContractsController,
     DeclaredContract,
+    ContractsController,
+    DeclaredSierraClass,
 )
-from protostar.compiler import ProjectCompiler
-
-from .callable_hint_local import CallableHintLocal
-
-
-class DeclareCheatcodeProtocol(Protocol):
-    def __call__(
-        self,
-        contract: str,
-        *args: Any,
-    ) -> DeclaredContract:
-        ...
 
 
 class DeclareHintLocal(CallableHintLocal):
-    def __init__(
-        self,
-        project_compiler: ProjectCompiler,
-        contracts_controller: ContractsController,
-    ):
+    def __init__(self, contracts_controller: ContractsController):
         self._contracts_controller = contracts_controller
-        self._project_compiler = project_compiler
 
     @property
     def name(self) -> str:
         return "declare"
 
-    def _build(self) -> Any:
+    def _build(self) -> Callable:
         return self.declare
 
     def declare(self, contract: str) -> DeclaredContract:
-        compiled_contract = (
-            self._project_compiler.compile_contract_from_contract_identifier(contract)
-        )
-        declared_class = asyncio.run(
-            self._contracts_controller.declare_contract(compiled_contract)
+        # TODO add this
+        # compiled_contract_sierra =  ...
+        contract_class = make_contract_class("")
+
+        compiled_contract_casm = compile_starknet_contract_from_path(
+            input_path=Path(contract)
         )
+        assert compiled_contract_casm is not None
+        compiled_class_hash = compute_class_hash(compiled_contract_casm)
 
-        assert declared_class
-        class_hash = declared_class.class_hash
+        declared_class: DeclaredSierraClass = asyncio.run(
+            self._contracts_controller.declare_sierra_contract(
+                contract_class=contract_class,
+                compiled_class_hash=compiled_class_hash,
+            )
+        )
 
         self._contracts_controller.bind_class_hash_to_contract_identifier(
-            class_hash=class_hash,
+            class_hash=declared_class.class_hash,
             contract_identifier=contract,
         )
 
-        return DeclaredContract(class_hash)
+        return DeclaredContract(class_hash=declared_class.class_hash)
+
+
+def make_contract_class(sierra_contract: str) -> ContractClass:
+    return ContractClass.loads(sierra_contract)
+
+
+def compute_class_hash(casm_contract: str) -> int:
+    return compute_compiled_class_hash(parse_casm(json.loads(casm_contract)))
```

Comment:
Those functions seem a bit too anemic, can we avoid it? It is completely understandable what they do from their body. Introducing additional functions brings unnecessary mental overhead when jumping around code